cms/djangoapps/api/v1/tests/test_views/test_course_runs.py

```diff
@@ -138,44 +138,76 @@ def test_update_with_invalid_user(self):
         assert response.status_code == 400
         assert response.data == {'team': [{'user': ['Object with username=test-user does not exist.']}]}
 
-    def test_partial_update(self):
+    def test_role_get_or_create(self):
+        """
+        Verify that the api gets or creates course access role
+        """
+        instructor_role = 'instructor'
+        start = datetime.datetime.now(pytz.UTC).replace(microsecond=0)
+        new_user = UserFactory()
+        course_run = CourseFactory(start=start, end=None, enrollment_start=None, enrollment_end=None, self_paced=False)
+        assert CourseAccessRole.objects.filter(course_id=course_run.id).count() == 0
+        data = {
+            'team': [
+                {
+                    'user': new_user.username,
+                    'role': instructor_role,
+                },
+            ],
+            'pacing_type': 'self_paced',
+        }
+        url = reverse('api:v1:course_run-detail', kwargs={'pk': str(course_run.id)})
+        response = self.client.patch(url, data, format='json')
+        assert response.status_code == 200
+        self.assert_access_role(course_run, new_user, instructor_role)
+        self.assert_course_access_role_count(course_run, 1)
+
+        # The same request again should only get the instructor and not change anything
+        url = reverse('api:v1:course_run-detail', kwargs={'pk': str(course_run.id)})
+        response = self.client.patch(url, data, format='json')
+        assert response.status_code == 200
+        self.assert_access_role(course_run, new_user, instructor_role)
+        self.assert_course_access_role_count(course_run, 1)
+
+        course_run = modulestore().get_course(course_run.id)
+        assert course_run.self_paced is True
+        self.assert_course_run_schedule(course_run, start, None, None, None)
+
+    def test_multiple_roles(self):
+        """
+        Verify that the api creates multiple roles if a user already has one role
```

Comment:
something like? => `Tests that update course run handles multiple instructors correctly`